src/app/reports/reportsTable/getReportsData.ts

```diff
@@ -0,0 +1,103 @@
+import {
+    ReportsSortState,
+    ReportsTableRow,
+    GetReportsCountReturnType,
+    GetReportsReturnType,
+} from "./types";
+import { Supabase } from "@/supabaseUtils";
+
+const getReportsDataAndCount = async (
+    supabase: Supabase,
+    startIndex: number,
+    endIndex: number,
+    sortState: ReportsSortState,
+    abortSignal: AbortSignal
+): Promise<GetReportsReturnType> => {
+    let query = supabase.from("reports").select("*");
+
+    if (sortState.sortEnabled && sortState.column.sortMethod) {
+        query = sortState.column.sortMethod(query, sortState.sortDirection);
+    } else {
+        query.order("week_commencing", { ascending: false });
+    }
+
+    query = query.range(startIndex, endIndex);
+    query = query.abortSignal(abortSignal);
+
+    const { data: reports, error: reportsError } = await query;
+
+    if (reportsError) {
+        if (abortSignal.aborted) {
+            const logId = await logInfoReturnLogId("Aborted fetch: report table", {
+                error: reportsError,
+            });
+            return { error: { type: "abortedFetchingReportsTable", logId }, data: null };
+        }
+
+        const logId = await logErrorReturnLogId("Error with fetch: reports table", {
+            error: reportsError,
+        });
+        return { error: { type: "failedToFetchReportsTable", logId }, data: null };
+    }
+
+    const reportsData: ReportsTableRow[] = reports.map((report) => {
+        return {
+            weekCommencing: report.week_commencing ?? "",
+            total: report.total_parcels ?? 0,
+            familySize1: report.family_size_1 ?? 0,
+            familySize2: report.family_size_2 ?? 0,
+            familySize3: report.family_size_3 ?? 0,
+            familySize4: report.family_size_4 ?? 0,
+            familySize5: report.family_size_5 ?? 0,
+            familySize6: report.family_size_6 ?? 0,
+            familySize7: report.family_size_7 ?? 0,
+            familySize8: report.family_size_8 ?? 0,
+            familySize9: report.family_size_9 ?? 0,
+            familySize10Plus: report.family_size_10_plus ?? 0,
+            totalWithPets: report.total_with_pets ?? 0,
+            cat: report.cat ?? 0,
+            dog: report.dog ?? 0,
+            catAndDog: report.cat_and_dog ?? 0,
+        };
+    });
+
+    const { data: reportsCount, error: reportCountError } = await getReportsCount(
+        supabase,
+        abortSignal
+    );
+    if (reportCountError) {
+        return { data: null, error: reportCountError };
+    }
+
+    return { error: null, data: { reportsData, count: reportsCount } };
+};
+
+const getReportsCount = async (
+    supabase: Supabase,
+    abortSignal: AbortSignal
+): Promise<GetReportsCountReturnType> => {
+    let query = supabase.from("reports").select("*", { count: "exact", head: true });
+
+    query = query.abortSignal(abortSignal);
+
+    const { count, error: reportsError } = await query;
```

Comment:
IntelliJ/Typescript seems to know it's number | null, presumably as it's from a count query. I could give it an alias reportsCount or something like that but not sure if that adds anything? Thanks :)

Reply:
Think I like how it was done above, just a bit more consistent
```
const { data: reports, error: reportsError } = await query;
```
just change reports to count

Reply:
I'm a little confused -> do you mean to replace it with this:

`const { data: count, error: reportsError } = await query;`

I think this isn't equivalent to what I have as that's setting the data variable to count, rather than using count variable.

The query is `const query = supabase.from("reports").select("*", { count: "exact", head: true });` which returns data, count and error. Data would be all elements of the table, while count is the number of them - so they wouldnt' do the same thing. Lmk if I'm misunderstanding :)